library/library.mli

```diff
@@ -80,5 +76,10 @@ val locate_qualified_library :
 
 *)
 
+(** {6 ... }
+    Require = load in the environment + open (if the optional boolean
+    is not [None]); mark also for export if the boolean is [Some true] *)
+val require_library_from_dirpath : mode:require_mode -> (DirPath.t * string) list -> bool option -> unit
```

Comment:
Please avoid these long lines, I like `ocamlformat` style which is become quite standard in the larger OCaml ecosystem.

Reply:
Also the doc comment should be fixed I think.

Reply:
By "fixed" you mean to also say that "mode" defines what should be loaded? Maybe a better place for such a description would be near the declaration of `require_mode`, no?

Reply:
By fixed I meant that the `ocamldoc` is not correct, shouldn't have a section and should use the standard presentation format [function arg1 arg2 ...]

But indeed after reading the PR I realize we also need proper documentation, otherwise the situation here un-understandable.

Reply:
Note that this PR doesn't introduce or change the faulty comment.

I'll try to improve the documentation, though.

Reply:
> Note that this PR doesn't introduce or change the faulty comment.

This PR modifies the function so for sure has to update the doc comment. I know it is a pain but it also modifies critical rules, in particular w.r.t. build system so I'm much afraid that the new rules have to be properly documented before we let this in.

It is true that the rules should be already be documented in the past, but the fact that in the past we had laxer standards doesn't mean today we have to keep them.

Reply:
Sure, no pb, I'm already writing the new rules :)